### PR TITLE
better parse rpmbuild output

### PR DESCRIPTION
**Pull request by evgeni**
recent rpmbuild versions will output a lot of data on stdout and stderr, which can confuse our parsing. ignoring stderr doesn't completely help, so instead lets find the exact line saying "Wrote: some.src.rpm" and take that

### CODE REVIEW

File: obal/data/modules/srpm.py [hunk 1]
```diff
@@ -137,11 +137,16 @@ def main():
...
         except subprocess.CalledProcessError as error:
             module.fail_json(msg='Failed to srpm build', command=' '.join(command), output=error.output)
 
-        result = result.split("Wrote: ")[-1].rstrip()
+        result = None
+        for line in command_output.splitlines():
+            if line.startswith("Wrote: "):
+                result = line.split("Wrote: ")[-1].rstrip()
+        if not result:
+            module.fail_json(msg='Failed to find built SRPM', command=' '.join(command))
```
Comment (evgeni) [inline obal/data/modules/srpm.py:149]: pylint complains my code is too complicated :roll_eyes:

File: obal/data/modules/srpm.py [hunk 2]
```diff
@@ -137,11 +137,16 @@ def main():
...
-            result = run_command(command)
+            command_output = run_command(command)
         except subprocess.CalledProcessError as error:
             module.fail_json(msg='Failed to srpm build', command=' '.join(command), output=error.output)
 
-        result = result.split("Wrote: ")[-1].rstrip()
+        result = None
+        for line in command_output.splitlines():
+            if line.startswith("Wrote: "):
+                result = line.split("Wrote: ")[-1].rstrip()
```
Comment (ekohl) [inline obal/data/modules/srpm.py:147]: Should it break after this?

You can then also consider the `else` statement for `for`, though I know plenty of people don't like that.

Comment (evgeni) [reply obal/data/modules/srpm.py:146]: Done. But yes, `for/else` ain't my biggest friend.

Comment (ekohl) [reply obal/data/modules/srpm.py:146]: I had thought you'd move the `module.fail_json` into `else`. Though I now realize that possibly you also wanted to capture empty strings. If you're not moving `fail_json` into `else`, I'd say that your previous version where you first declared `result = None` before the loop was better (but with `break` as an optimization).

**Review by ekohl — APPROVED** [review]
Looks good to me, but the build is rather red.

**Comment by evgeni** [comment]
@ekohl this had more changes than just a limits bump, mind re-reviewing?

**Review by ekohl — APPROVED** [review]

**Pull request merged**